### ISSUE TITLE
test_passthroughfs sometimes fails due to timestamp difference

### ISSUE THREAD

**Issue by ThomasWaldmann** [libfuse/pyfuse3#57]
Seen on github actions CI:
```
Traceback (most recent call last):
  File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 124, in test_passthroughfs
    tst_passthrough(src_dir, mnt_dir)
  File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 412, in tst_passthrough
    assert_same_stats(src_name, mnt_name)
  File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 438, in assert_same_stats
    assert  v1 == v2, 'Attribute {} differs by {} ({} vs {})'.format(
AssertionError: Attribute st_mtime_ns differs by 4000074 (1654852943987067641 vs 1654852943983067567)
assert 1654852943987067641 == 1654852943983067567
```

**Comment by ThomasWaldmann** [comment]
Still happens, just have seen this with github CI for 3.2.2 on python 3.10 (tests passed by just retrying it).

**Comment by ThomasWaldmann** [comment]
```
______________________________ test_passthroughfs ______________________________
Traceback (most recent call last):
   File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 124, in test_passthroughfs
    tst_passthrough(src_dir, mnt_dir)
  File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 414, in tst_passthrough
    assert_same_stats(src_name, mnt_name)
  File "/home/runner/work/pyfuse3/pyfuse3/test/test_examples.py", line 440, in assert_same_stats
    assert  v1 == v2, 'Attribute {} differs by {} ({} vs {})'.format(
AssertionError: Attribute st_mtime_ns differs by 3999985 (1691339543559117222 vs 1691339543555117237)
assert 1691339543559117222 == 1691339543555117237
```

**Comment by ThomasWaldmann** [comment]
@Nikratio I am wondering: is this just a cosmetic issue due to timestamp granularity of different filesystems? 4ms seems a bit much for that. Or is it a bug? What do you think?

**Comment by Nikratio** [comment]
I'm not sure what's going on here. FUSE uses ns-granularity to report the timestamp from the underlying filesystem, so there should be no loss of accuracy.

**Comment by Nikratio** [comment]
To debug this, I'd add a `sleep` prior to the comparison and see if that makes the difference go up.

**Comment by Nikratio** [comment]
What's happening here is the following:

- If the FUSE (kernel-level) writeback cache is enabled, the filesystem daemon cannot be trusted by the kernel to produce accurate mtime and ctime values, because there may be pending writes in the cache that have not been flushed to the daemon.
- Therefore, the kernel still calls the getattr() handler, but ignores the mtime and ctime values, and instead maintains them internally.
- When a file is closed, the kernel communicates the correct mtime and ctime values to the filesystem daemon through an extra setattr() call.
- Therefore, the (correct) mtime value that a userspace program gets from the kernel for the FUSE filesystem will not agree with the (also correct) mtime value reported by the underlying filesystem as long as the file is open.
- Once the file has been closed, the mtime values agree, but the filesystem daemon has updated this with an utimens() call which has resulted in ctime change, so now the ctime values do not agree.

In other words, the bug here is in the test, not the filesystem or the Python bindings.
